grammars/clojure.cson

```diff
@@ -277,7 +277,8 @@
     'name': 'meta.expression.clojure'
     'patterns': [
       {
-        'begin': '(?<=\\()(ns|def|def-|defn|defn-|defvar|defvar-|defmacro|defmacro-|deftest)\\s+'
+        # ns, declare and everything that starts with def* or namespace/def*
+        'begin': '(?<=\\()(ns|declare|def[\\w\\.\\-\\_\\:\\+\\=\\>\\<\\!\\?\\*\\d]*|[\\w\\.\\-\\_\\:\\+\\=\\>\\<\\!\\?\\*][\\w\\.\\-\\_\\:\\+\\=\\>\\<\\!\\?\\*\\d]*/def[\\w\\.\\-\\_\\:\\+\\=\\>\\<\\!\\?\\*\\d]*)\\s+'
```

Comment:
The majority of the symbols in the character class don't need to be escaped (in fact, everything but `\\w` and `\\d` don't need to be escaped, and you can put the `-` at the end of the character class.

In addition, if I'm reading this correctly, something like `def+!.?abc8<>:` is a valid macro?